packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -158,9 +157,10 @@ abstract class GlobalKey<T extends State<StatefulWidget>> extends Key {
     assert(() {
       if (_registry.containsKey(this)) {
         assert(element.widget != null);
-        assert(_registry[this].widget != null);
-        assert(element.widget.runtimeType != _registry[this].widget.runtimeType);
-        _debugIllFatedElements.add(_registry[this]);
+        final Element oldElement = _registry[this]!;
+        assert(oldElement.widget != null);
+        assert(element.widget.runtimeType != _registry[this]!.widget.runtimeType);
+        _debugIllFatedElements.add(_registry[this]!);
```

Comment:
we should probably use oldElement throughout here (or not at all, but i see the benefit of avoiding the `!`s)

Reply:
Done